client/my-sites/earn/memberships/index.tsx

```diff
@@ -88,10 +121,14 @@ function MembershipsSection( { query } ) {
 	}, [ query, site ] );
 
 	function renderEarnings() {
+		if ( ! site ) {
+			return <LoadingPlaceholder />;
+		}
```

Comment:
I had to add several checks like this because the QueryMembershipsEarnings requires a valid site ID.

Reply:
@simison - I'll plan to add that as a piece of the broader update of the Earn code.